### PR TITLE
feat/AB#74580_add_pagination_main_users_in_backoffice

### PR DESCRIPTION
**Pull request by RenzoPrats**
# Description

Added pagination to main users table in back-office.

## Useful links

[ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/74580)
[frontend](https://github.com/ReliefApplications/oort-frontend/pull/1822)

## Type of change

- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

It has been tested accessing the users table in the application and testing if the pagination is working correctly.

## Screenshots

Peek 12-09-2023 17-14

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by estelafs — CHANGES_REQUESTED** [review]
Updated the base branch to the new ticket base branch (2.x.x).
Requested changes: table pagination isn't updated when adding or deleting users and the filter only considers the visible contents of the table instead of refetching using and use the filters params. Requested changes in the frontend as well.

**Review by estelafs — APPROVED** [review]

**Review by Matthis-M-ReliefApps — APPROVED** [review]
Seems good to me

File: src/schema/query/users.query.ts [hunk 1]
```diff
@@ -1,38 +1,133 @@
...
+  afterCursor?: string;
 };
 
 /**
  * List back-office users if logged user has admin permission.
  * Throw GraphQL error if not logged or not authorized.
  */
 export default {
-  type: new GraphQLList(UserType),
+  type: UserConnectionType,
```
Comment (AntoineRelief) [inline src/schema/query/users.query.ts:65]: @estelafs @RenzoPrats @Matthis-M-ReliefApps 
you all missed that this query is also used by a surveyjs question, so this is breaking the question 😅

**Comment by AntoineRelief** [comment]
closing and putting most of changes in this one:
https://github.com/ReliefApplications/ems-backend/pull/866

File: src/schema/query/users.query.ts (hunk 1 above)
Comment (Matthis-M-ReliefApps) [reply src/schema/query/users.query.ts:65]: Oh damn, completely missed that indeed :open_mouth: will be more careful about where the queries are used next time

**Pull request closed without merging**